### ISSUE TITLE
Add sprite movement helper functions to GameBackbone

### ISSUE THREAD

**Issue by lavinrp** [lavinrp/GameBackbone#91]
Every client of backbone will need to be able to move sprites.
Add helper functions to backbone to facilitate this.

**Comment by lavinrp** [comment]
Clients can use std::async to thread these out. Dont worry about making them on different threads.

**Comment by lavinrp** [comment]
Also make a class for translating between NavGrid coordinates and window coordinates.

**Comment by lavinrp** [comment]
@bauerbp @gyurgyma @malexmc 
There is behavior in the current implementation that may be unacceptable. 

> The sprites will stop after reaching each point in the path even if they are capable of moving farther.

This is because the path following just calls moveStepTowardsPoint. If we want sprites to start progress towards next point if they have enough movement left then a significant functionality / testing re-write will be required.

I'm not fully opposed to this.

**Comment by lavinrp** [comment]
Need to add CompoundSprite movement functionality.

**Comment by lavinrp** [comment]
Blocked by #103

**Comment by lavinrp** [comment]
Resuming work.

**Comment by lavinrp** [comment]
@malexmc @bauerbp 
I'm going to need some help with this one to get it done quickly.

the following test suites need to be implemented. 

1. moveCompoundSpriteAlongPathTests
2. bulkMoveCompoundSpriteAlongPathTests
3. moveCompoundSpriteStepTowardsPointTests
4. bulkMoveCompoundSpriteStepTowardsPointTests

It shouldn't be too difficult to implement these test suites since there is a matching test suite for the sprite movement helpers (these are all compound sprite movement helpers).

Would you guys mind:
1. picking a test suite to help implement
2. creating a branch (not a new feature branch) to fill out the suite
3. comment on the issue with which one of the four you are working on

**Comment by lavinrp** [comment]
working on part 1.

**Comment by lavinrp** [comment]
working on part 2.

**Comment by lavinrp** [comment]
This is going to need some refactoring. Hold off on testing for now.

**Comment by lavinrp** [comment]
Bulk functions were more trouble than there were worth. Would require too much change to make them work as part of GB.

Bulk implementations of these functions can still be added by any clients.

**Comment by gyurgyma** [comment]
Merged into Dev.
